packages/sit-onyx/src/components/OnyxMenuItem/OnyxMenuItem.vue

```diff
@@ -0,0 +1,67 @@
+<script setup lang="ts">
+import { inject } from "vue";
+import OnyxListItem from "../OnyxListItem/OnyxListItem.vue";
+import { MENU_BUTTON_ITEM_INJECTION_KEY } from "./types";
+import { type OnyxMenuItemProps } from "./types";
+
+const props = defineProps<OnyxMenuItemProps>();
+const emit = defineEmits<{
+  /**
+   * Emitted when the menu item is clicked (via click or keyboard).
+   */
+  click: [];
+}>();
+
+const menuButton = inject(MENU_BUTTON_ITEM_INJECTION_KEY);
+</script>
+<template>
+  <OnyxListItem
+    :selected="props.active"
+    :active="props.active"
+    v-bind="menuButton?.listItem"
+    :color="props.color"
+    :disabled="props.disabled"
+    class="onyx-menu-item"
+  >
+    <component
+      :is="props.href ? 'a' : 'button'"
+      :class="{
+        'onyx-list-item': !props.href,
```

Comment:
We must use bem classes of the component itself, the bem class of another component must not be used. Otherwise we will have unexpected side-effects in this component when editing the `OnyxListItem`.
```suggestion
```

Reply:
As a trick you could apply `display: contents` to the `anchor` and `button` element.

Reply:
display: contents messes up our keyboard navigation

Reply:
Then we need to override the style manually